DistroLauncher/PatchConfig.h

```diff
@@ -0,0 +1,116 @@
+    /// It implements a callable interface to visit the [PatchFnPtr] variant and apply the patching action.
+    /// That's why this is the Patcher.
+    /// It's creation and destruction are meant to be cheap enough to allow creation of instances on demand, one
+    /// instance per patch to be applied.
+    class Patcher
+    {
+      private:
+        /// The absolute file path to the config file to be patched already translated to Windows paths.
+        std::filesystem::path translatedFilePath;
+
+        /// The stream a patching function must write into. Always starts empty.
+        std::stringstream modified;
+
+        /// Applies the contents of the file stream to the destination file path inside the distro. It assumes it must
+        /// write the file, thus must be private so that precondition is ensured prior to calling this method.
+        bool commit();
+
+      public:
+        /// Creates a new Patcher instance, storing [linuxFile] translated to Windows paths according to a [pathPrefix].
+        Patcher(std::filesystem::path pathPrefix, std::filesystem::path linuxFile);
+
+        /// Movable but not copyable due the underlying stream, which is not copyable.
+        Patcher() = default;
+        Patcher(Patcher&& other) = default;
+        ~Patcher() = default;
+        Patcher(const Patcher& other) = delete;
+        Patcher operator=(const Patcher& other) = delete;
+
+        /// Calls a [CreateFn] patching function. If the supplied callable succeeds, the modifications on the output
+        /// file stream are commited to the original config file.
+        inline bool operator()(CreateFn* p)
+        {
+            if (!p(modified)) {
+                return false;
+            }
+            return commit();
+        }
+
+        /// Calls a [ModifyFn] patching function passing a read-only file stream pointing to the [configfilePath]. If
+        /// the supplied callable succeeds, the modifications on the output file stream are commited to the original
+        /// config file.
+        inline bool operator()(ModifyFn* p)
+        {
+            // let the destructor close it automatically.
+            std::ifstream original{translatedFilePath};
+            if (original.fail()) {
+                return false;
+            }
+            if (!p(original, modified)) {
+                return false;
+            }
+            return commit();
+        }
+
+        // Observers (made public mainly for testing, allowing observing if certain invariants hold.
+        // They are likely to never be called in production code).
+
+        /// Returns a copy of the translatedFilePath defined during construction.
+        std::filesystem::path translatedPath() const
+        {
+            return translatedFilePath;
+        }
+    };
+
+    /// A data structure defining a patch to be done to a specific configuration file.
+    struct PatchConfig
```

Comment:
That's an interesting suggestion. My original intent was to make the `PatchConfig` lower level and dummy. But I guess your suggestion makes more sense, since touching files is the lowest level and the declarative composition abstracts on top of it.